storage-proofs-porep/src/stacked/vanilla/proof.rs

```diff
@@ -380,16 +393,31 @@ impl<'a, Tree: 'static + MerkleTreeTrait, G: 'static + Hasher> StackedDrg<'a, Tr
                         );
 
                         {
+                            /*
+                            //FIXME: For testing, this adds a bug on purpose
+                            let labeled_node =
+                                *rcp.c_x.get_node_at_layer(if layer >= num_layers {
+                                    1
+                                } else {
+                                    layer + 1
+                        })?;
+                             */
                             let labeled_node = *rcp.c_x.get_node_at_layer(layer)?;
                             let replica_id = &pub_inputs.replica_id;
                             let proof_inner = proof.clone();
+                            let invalid_encoding_proof_inner = Arc::clone(&invalid_encoding_proof);
                             scope.execute(move || {
-                                assert!(
-                                    proof_inner.verify(replica_id, &labeled_node),
-                                    "Invalid encoding proof generated at layer {}",
-                                    layer,
-                                );
-                                trace!("Valid encoding proof generated at layer {}", layer);
+                                if !proof_inner.verify(replica_id, &labeled_node) {
+                                    let mut invalid = invalid_encoding_proof_inner.lock().expect("foo");
```

Comment:
The "foo" should be something like "cannot acquire lock" or "lock to be acquired". (The same for other "foo" below).